### PR TITLE
Adding plots

### PR DESCRIPTION
**Pull request by AlexDeLos**

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def get_arguments():
     parser.add_argument("-l", "--learning_rate", default=1e-4)
     parser.add_argument("-w", "--weight_decay", default=0.05)
     parser.add_argument("--patience", default=40)
+    parser.add_argument("--plot_node_error", default=True)
```
Comment (francescoopiccoli) [inline src/utils.py:39]: the flag is not working, even if I set to false, it still plots, you should as I did for normalise one line below

File: src/plot_utils.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    timestamp = pd.Timestamp.now().strftime("%Y-%m-%d %H:%M")
+
+    model_name = "distance_plot" + "_" + model.class_name + "_" + str(timestamp)
+    plt.savefig(f"plots/{model_name}.png")
+
+
+def get_distance_loss(out,labels,data):
+    res = [0]
+    norm = [0]
```
Comment (francescoopiccoli) [inline src/plot_utils.py]: what are res and norm, why they are initialised as [0]

File: src/plot_utils.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        if dis != -1:
+            if dis > len(res)-1:
+                res = res + [0]*(dis-len(res) +1)
+                norm = norm + [0]*(dis-len(norm) +1)
+            norm[dis] += 1
+            res[dis] += th.sum(th.abs(out[i]-labels[i])).item()
+    for i in range(len(res)):
+        if norm[i] != 0:
+            res[i] = res[i]/norm[i]
+    return res, len(res)
```
Comment (francescoopiccoli) [inline src/plot_utils.py]: maybe some comments or a more explicit variable naming makes the code easier to read

File: src/plot_utils.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            result[el] = max_distance
+    return result
+
+
+def get_neighbors(data, node):
+    neighbors = set()
+    edges = data.edge_index
+
+    #assume they are ordered
+    #broken:
```
Comment (francescoopiccoli) [inline src/plot_utils.py]: @AlexDeLos is broken an outdated comment?

**Review by francescoopiccoli — COMMENTED** [review]
I would improve variable naming in plot_utils and add some comments to explain some steps

File: src/plot_utils.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            norm[dis] += 1
+            res[dis] += th.sum(th.abs(out[i]-labels[i])).item()
+    for i in range(len(res)):
+        if norm[i] != 0:
+            res[i] = res[i]/norm[i]
+    return res, len(res)
+
+
+def MES_loss(cur,out,label):
+    return th.add(cur+th.abs(out-label))
```
Comment (francescoopiccoli) [inline src/plot_utils.py]: @AlexDeLos isn't it adding twice using .add and also "+"?

File: .gitignore [hunk 6]
```diff
@@ -178,3 +178,4 @@ trained_models/
 
 loaded_data/
 plots/
+distance_plot.png
```
Comment (francescoopiccoli) [inline .gitignore]: @AlexDeLos the plots are now saved in plots/ folder, no need to add the png

File: src/utils.py [hunk 7]
```diff
@@ -36,8 +36,8 @@ def get_arguments():
     parser.add_argument("-l", "--learning_rate", default=1e-4)
     parser.add_argument("-w", "--weight_decay", default=0.05)
     parser.add_argument("--patience", default=40)
-    parser.add_argument("--plot_node_error", action="store_true", default=False)
-    parser.add_argument("--normalize", action="store_true", default=False)
+    parser.add_argument("--plot_node_error", action="store_true", default=True)
+    parser.add_argument("--normalize", action="store_true", default=True)
```
Comment (francescoopiccoli) [inline src/utils.py]: @AlexDeLos Changing default to true makes it impossible to set the value to False, when calling from command line

File: src/plot_utils.py [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    """
+    out = model(batch)
+    distance_loss,len = get_distance_loss(out,batch.y,batch)
+    plt.bar(list(range(0,len)), distance_loss, color ='maroon')
+    plt.title("Error with distance from the generator")
+    plt.ylabel("Error")
+    plt.xticks(range(0,len))
+    plt.xlabel("Nodes away from the generator the node was located")
+    plt.show()
+    plt.savefig("distance_plot.png")
```
Comment (francescoopiccoli) [inline src/plot_utils.py]: @AlexDeLos save fig needs to be before show() otherwise its gonna save a blank image, also my changes have been deleted, I had added a better naming for the png file please check my previous commits

**Review by francescoopiccoli — DISMISSED** [review]

File: src/plot_utils.py (hunk 3 above)
Comment (AlexDeLos) [reply src/plot_utils.py]: Added documentation
